### ISSUE TITLE
Design: Update org model table search bar so the form field label is visible

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2824]
### Story

The search bar for the domains and domain requests table does not have a visual label when user input is in the form field. This is a failure section 508 trusted tester ID 3.3.2-label-provided.

### Acceptance Criteria

- [x] Redesign the search bar component with a label that is visible at all times.
- [x] Review designs with design team, product, and dev team.
- [x] Create a development ticket.

### Issue Links

🔄 Relates to #2535
